### PR TITLE
Make thread pool size configurable for parallel backend queries

### LINKED ISSUE #2473: Make Backend thread pool size configurable
**Describe the feature:**
Making Backend thread pool size configurable. Based on my understanding, when executing edgeStoreMultiQuery, this thread pool in Backend is used and each getSlice will occupy one thread. As there are only # core * 2 threads,  the overall execution parallelism is limited by this size of the pool. Should allow the size customizable.

**Describe a specific use case for the feature:**
[If possible add a description of the specific use case.]

### PR DESCRIPTION
**Pull request by porunov**
Fixes #2473

Signed-off-by: Oleksandr Porunov <alexandr.porunov@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

File: janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/GraphDatabaseConfigurationInstanceIdTest.java [hunk 1]
```diff
@@ -100,4 +103,19 @@ public void instanceIdShouldEqualHostnamePlusSuffix() throws UnknownHostExceptio
         assertEquals(graph.openManagement().getOpenInstances().toArray()[0], Inet4Address.getLocalHost().getHostName() + "1");
 				graph.close();
     }
+
+    @Test
+    public void shouldCreateCustomBackendThreadPoolSize() throws UnknownHostException {
```
Comment (li-boxuan) [inline janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/GraphDatabaseConfigurationInstanceIdTest.java:108]: Are you able to run this test locally? I got `java.lang.IllegalArgumentException: Could not find implementation class: org.janusgraph.diskstorage.inmemory.InMemoryStoreManager` when I try running this test class. I suspect it is not picked up by CI at all.

Comment (porunov) [reply janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/GraphDatabaseConfigurationInstanceIdTest.java:108]: You are right. Thank you for this catch! I will fix it in a separate PR

Comment (li-boxuan) [reply janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/GraphDatabaseConfigurationInstanceIdTest.java:108]: I guess you will write a test in another place, e.g. JanusGraphTest.java, right?

Comment (porunov) [reply janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/GraphDatabaseConfigurationInstanceIdTest.java:108]: Kind of yes. I'm working on it right now but it uncovered some mistakes in these tests and a bug. Fixing it. I will provide a PR today

Comment (porunov) [reply janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/GraphDatabaseConfigurationInstanceIdTest.java:108]: Oh sorry. No, I will not write a test in a separate place. I want first fix these tests and then rebase this PR

Comment (li-boxuan) [reply janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/GraphDatabaseConfigurationInstanceIdTest.java:108]: Got it, that sounds fair

Comment (porunov) [reply janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/GraphDatabaseConfigurationInstanceIdTest.java:108]: Fixed tests here #2697 except the one which is now tracked here #2696

Comment (porunov) [reply janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/GraphDatabaseConfigurationInstanceIdTest.java:111]: @li-boxuan I have rebased this PR against the latest master commit, so this test should run now

**Review by li-boxuan — APPROVED** [review]
LGTM, thank you!

**Pull request merged**